lib/spack/spack/test/conftest.py

```diff
@@ -744,34 +817,7 @@ def module_configuration(monkeypatch, request):
         spack.paths.test_path, 'data', 'modules', writer_key
     )
 
-    def _impl(filename):
```

Comment:
Are methods defined in a method bad for pickle? I know classes defined at non-module scope are a problem, but I thought you could pickle functions defined at any scope.

Reply:
> Are methods defined in a method bad for pickle?

Yes

Reply:
We should probably add something to that effect to the developer guide. I don't think it should be a problem to ban nested methods and nested functions from our style guide.